src/app/components/categories-table/categories-table.component.html

```diff
@@ -24,19 +24,20 @@ <h1>Administrate categories</h1>
           <th mat-header-cell  *matHeaderCellDef class="table-header-text"> Actions </th>
           <td mat-cell *matCellDef="let element" >
             <button mat-button (click)="editCategory(element.id)"><mat-icon>edit</mat-icon></button>
-            <button mat-button (click)="deleteCategory(element.id)"><mat-icon>delete</mat-icon></button>
+            <button mat-button (click)="deleteCategory(element.id)" (click)="$event.stopPropagation()"><mat-icon>delete</mat-icon></button>
```

Comment:
You can have them both under the same click event

(click)="deleteCategory(element.id); $event.stopPropagation()"